### PR TITLE
#4241 fix: Compose recipient box UI bug

### LINKED ISSUE #4241: Compose message box UI bug: updating recipients
It may relates to the "password-protected message" feature. Please look at the attached video. Updating recipients sometimes can have unexpected results (some recipients can have PGP but shoudn't, and vice versa). To reproduce just need to play around different recipients. Also the updating process looks frozen(Infinite progress)

https://user-images.githubusercontent.com/2863246/148768108-a84a37e2-d80c-4b6b-b1b0-ac53de1680d7.mp4

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR fixed Compose recipient box UI bug

close #4241 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by ioanmo226** [comment]
@DenBond7 Can you help me test if the issue has been resolved on your end too with this pull request?
If you're not sure how to build the extension, just let me know and I'll be happy to help.

**Comment by DenBond7** [comment]
@ioanmo226 Sure. Please share details on how I can test it(how to build the extension). Just in case, I use Ubuntu 20.04

**Comment by ioanmo226** [comment]
@DenBond7 Could you follow readme setup introduction guide first and let me know if you have questions/troubles?

**Comment by ioanmo226** [comment]
Actually, you just need to clone repository, move to current PR and then run `npm install` and run `npm run build`

**Comment by DenBond7** [comment]
@ioanmo226 I've compiled the project. How can I install the extension?

**Comment by ioanmo226** [comment]
which browser are you using?

**Comment by DenBond7** [comment]
chrome

**Comment by ioanmo226** [comment]
Please follow those steps.
1. Open your Chrome browser and navigate to chrome://extensions/.
2. If not already enabled, toggle on the Developer mode switch located at the top-right corner.
3. Click on the `Load Unpacked` button.
4. Browse to and select the appropriate folder, either `build/chrome-consumer` or `chrome-enterprise`.

Once completed, the extension will be successfully installed and ready to use.

**Comment by DenBond7** [comment]
Should I remove the production version before?

**Comment by ioanmo226** [comment]
@sosnovsky I think it'd be better to document above(↑) in readme?

**Comment by ioanmo226** [comment]
You don't need. Just disabling extension would do.
Or you may try with another user/browser.

**Comment by DenBond7** [comment]
By the way, it would be nice to have some markers to understand that I use a dev version. For now, for the dev version, I see the same info as the production version.

image

It will be better to see something like this one if possible.
image

**Comment by ioanmo226** [comment]
I agree with @DenBond7. @sosnovsky  what's your thought?

**Comment by DenBond7** [comment]
> Can you help me test if the issue has been resolved on your end too with this pull request?

@ioanmo226 I confirm. The issue is resolved by this PR.

**Comment by ioanmo226** [comment]
Thank you.

**Comment by sosnovsky** [comment]
> I think it'd be better to document ⬆️ in readme?

For sure, I thought we already has this info in README. Created https://github.com/FlowCrypt/flowcrypt-browser/issues/5016

> By the way, it would be nice to have some markers to understand that I use a dev version. For now, for the dev version, I see the same info as the production version.

Sounds good, it seems we can check if extension was loaded with `Load unpacked` and then add `-dev` suffix in status bar.
Created https://github.com/FlowCrypt/flowcrypt-browser/issues/5017

File: extension/chrome/elements/compose-modules/compose-recipients-module.ts [hunk 1]
```diff
@@ -1135,7 +1138,9 @@ export class ComposeRecipientsModule extends ViewModule<ComposeView> {
   };
 
   private generateRecipientId = (): string => {
-    return `recipient_${this.addedRecipients.length}`;
+    const recipientId = `recipient_${this.uniqueRecipientIndex}`;
+    this.uniqueRecipientIndex += 1;
+    return recipientId;
```
Comment (sosnovsky) [inline extension/chrome/elements/compose-modules/compose-recipients-module.ts:1143]: It seems unnecessary to store separate variable for unique index.

What about replacing it with
```js
private generateRecipientId = (): string => {
     const uniqueId = Math.floor(Math.random() * Date.now());
     return `recipient_${uniqueId}`;
}
```

Comment (ioanmo226) [reply extension/chrome/elements/compose-modules/compose-recipients-module.ts:1143]: Well, I think generating random id would be not good for testing.

Comment (sosnovsky) [reply extension/chrome/elements/compose-modules/compose-recipients-module.ts:1143]: Yes, you're right, then let's leave it as it is 👍

**Review by sosnovsky — APPROVED** [review]

**Pull request merged**